jsonrpc/server.go

```diff
@@ -205,6 +213,7 @@ type connection struct {
 	w         io.Writer
 	activated <-chan struct{}
 
+	// todo: guard for this in the code! don't depend on devs finding out about this!
```

Comment:
I don't think we should guard this because `connection` created upon websocket connection to one client, also `initialErr` is set during initial setup

Reply:
all update this now

Reply:
before i make the push.
I the idea was to make the error thread safe by wrapping in a mutex guarded error object.
that gets set after the inner lock has been acquired

```go
type ThreadSafeError struct {
 error
 lock
}

func (t *ThreadSafeError) SetError() {
 acquire lock
 set error
 release lock
}
```

Reply:
I'm not sure that if we needed we would need to do this anyway. Because error is an interface which in turn is just a pointer to a struct, every update of a pointer is atomic even if you don't guard it with mutexes

Reply:
oh cool!, I didn't know this!
do you have any go doc that describes this more.